pkg/service/managedwatchers.go

```diff
@@ -30,9 +30,12 @@ func (s *DeviceService) AddProvisionWatcher(watcher models.ProvisionWatcher) (st
 
 	_, ok := cache.Profiles().ForName(watcher.ProfileName)
 	if !ok {
-		errMsg := fmt.Sprintf("device profile %s doesn't exist for provision watcher %s", watcher.ProfileName, watcher.Name)
-		s.LoggingClient.Error(errMsg)
-		return "", errors.NewCommonEdgeX(errors.KindEntityDoesNotExist, errMsg, nil)
+		_, err := s.edgexClients.DeviceProfileClient.DeviceProfileByName(context.Background(), watcher.ProfileName)
```

Comment:
Why not put the profile in the cache here at this point? Will there not be other issues accessing the profile when the ProvisionWatcher detects a new device???

Or better yet, just put it in the cache during the start-up code that loads and pushes/pulls the profile to/from Core Metadata, like it was in V1. https://github.com/edgexfoundry/device-sdk-go/blob/main/internal/provision/profiles.go#L83-L90

This enables hybrid mode (command line with Core Metadata in docker) to work smoother for development purposes. It also reduces the complexities (amount of code changes in the PR)

Reply:
> Why not put the profile in the cache here at this point? Will there not be other issues accessing the profile when the ProvisionWatcher detects a new device???

I thought it just a convenient func to add the ProvisionWatcher to core-metadata like the 
 AddDevice func of manageddevices.go. 
https://github.com/edgexfoundry/device-sdk-go/blob/main/pkg/service/manageddevices.go#L27
The core-metadata will invoke the device service callback after adding the ProvisionWatcher.

> Or better yet, just put it in the cache during the start-up code that loads and pushes/pulls the profile to/from Core Metadata, like it was in V1.

It should be an enhancement to let the user predefine the ProvisionWatcher.

Reply:
@weichou1229, with your approach we will have grown to have 4 locations of code that are loading profiles from Core Metadata (3 of which are ignoring the returned profile):
1. The one above that you have added.
2. Here once the devices have been loaded: https://github.com/edgexfoundry/device-sdk-go/blob/5b7ab9a3ff9866f95cdc84e39a450aa5b3c9b9ea/internal/cache/init.go#L39
3. Here when a device is loaded: https://github.com/edgexfoundry/device-sdk-go/blob/5b7ab9a3ff9866f95cdc84e39a450aa5b3c9b9ea/pkg/service/manageddevices.go#L34
4. Here on startup: https://github.com/edgexfoundry/device-sdk-go/blob/main/internal/provision/profiles.go#L83
5. If the 4th one would simply put the profiles into cache, instead of ignoring the response the other 3 are not needed.

Reply:
@lenny-intel you are right.  I didn't notice that.  we should add it to cache whenever loading from core-metadata to prevent additional query.

Reply:
Added code to put the profiles into the cache when loading the predefined profile.